orte/mca/odls/alps/odls_alps_module.c

```diff
@@ -423,6 +423,11 @@ static int do_child(orte_odls_spawn_caddy_t *cd, int write_fd)
     sigprocmask(0, 0, &sigs);
     sigprocmask(SIG_UNBLOCK, &sigs, 0);
 
+    /* take us to the correct wdir */
+    if (NULL != cd->wdir) {
+        chdir(cd->wdir);
```

Comment:
I don't disagree strongly, although I think your request is already covered by the call to orte_util_check_context_cwd() (https://github.com/open-mpi/ompi/blob/master/orte/util/context_fns.c#L56).

Note that my addition is mirroring the existing code in the default ODSL module  (https://github.com/open-mpi/ompi/blob/master/orte/mca/odls/default/odls_default_module.c#L417).